pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -101,22 +101,22 @@ func (m topologyToMatchedTermCount) clone() topologyToMatchedTermCount {
 	return copy
 }
 
+// update updates the count of pods that meet the matching criteria in a particular topology.
+// This function assumes the node has the label keyed with `tk`.
 func (m topologyToMatchedTermCount) update(node *v1.Node, tk string, value int64) {
-	if tv, ok := node.Labels[tk]; ok {
-		pair := topologyPair{key: tk, value: tv}
-		m[pair] += value
-		// value could be negative, hence we delete the entry if it is down to zero.
-		if m[pair] == 0 {
-			delete(m, pair)
-		}
+	pair := topologyPair{key: tk, value: node.Labels[tk]}
```

Comment:
Sorry, on the second thought, can we change `update` fn to receive the label value like:

```
func (m topologyToMatchedTermCount) update(topologykey, topologyvalue string, value int64) {
```

Then, we reduce the num of lookup `node.Labels[tk]` here. Also, we can eliminate the future mistake of passing `node` which doesn't have the label keyed with `tk`

Reply:
Nice catch!

Changed the `update` function and lookup the `node.Labels` as little as possible, PTAL